--- lib/model/get_system_waypoints_traits_parameter.dart	2026-03-27 16:41:57.411306951 +0100
+++ lib/model/get_system_waypoints_traits_parameter.dart.patched	2026-03-27 16:42:06.901358562 +0100
@@ -12,14 +12,15 @@
 
 class GetSystemWaypointsTraitsParameter {
   /// Returns a new [GetSystemWaypointsTraitsParameter] instance.
-  GetSystemWaypointsTraitsParameter({
-  });
+  GetSystemWaypointsTraitsParameter() {
+    throw UnsupportedError("Bad OpenApi Generation");
+  }
 
   @override
-  bool operator ==(Object other) => identical(this, other) || other is GetSystemWaypointsTraitsParameter &&
+  bool operator ==(Object other) => identical(this, other) || other is GetSystemWaypointsTraitsParameter;
 
   @override
-  int get hashCode =>
+  int get hashCode => 0;
     // ignore: unnecessary_parenthesis
 
   @override
--- lib/model/warp_ship200_response_any_of1_data.dart	2026-03-27 15:08:23.476230611 +0100
+++ lib/model/warp_ship200_response_any_of1_data.dart.patched	2026-03-27 16:44:14.120055900 +0100
@@ -23,7 +23,7 @@
   bool destroyed;
 
   /// Navigation data (null when destroyed).
-  ModelNull nav;
+  ShipNav? nav;
 
   /// Fuel consumed before destruction.
   ShipFuel fuel;
@@ -93,7 +93,7 @@
 
       return WarpShip200ResponseAnyOf1Data(
         destroyed: mapValueOfType<bool>(json, r'destroyed')!,
-        nav: ModelNull.fromJson(json[r'nav'])!,
+        nav: ShipNav.fromJson(json[r'nav']),
         fuel: ShipFuel.fromJson(json[r'fuel'])!,
         message: mapValueOfType<String>(json, r'message')!,
       );
